lib/generator/files/track_files.rb

```diff
@@ -53,6 +27,16 @@ def exercise_path
         File.join(paths.track, 'exercises', exercise_name)
       end
 
+      # this method contains a LOT of magic text
+      def tests_template_filename
+        File.exist?(track_tests_template_filename) ? track_tests_template_filename :
+          File.join(paths.track, 'lib', 'generator', 'test_template.erb')
+      end
+
+      def track_tests_template_filename
+        @track_tests_template_filename ||= File.join(exercise_path, 'example.tt')
```

Comment:
Heh. I memoize once something hits 2. It's easier than trying to come up with some rule of thumb greater than 2 and far better than what I've seen in the wild, which is tons of methods that are run many times without memoization.